site/src/lib/api/blocks/r2.ts

```diff
@@ -143,9 +143,22 @@ const uploadBlockFilesToR2 = (
         `${remoteStoragePrefix}/${thingName}`,
       );
     }
-    const fileContents = fs.readFileSync(pathToThing).toString();
+    const fileContents = fs.readFileSync(pathToThing);
 
     const contentType = mime.lookup(thingName);
+    if (
+      contentType &&
+      contentType.includes("svg") &&
+      fileContents
+        .toString()
+        .match(
+          /(script|entity|onerror|onload|onmouseover|onclick|onfocus|foreignObject)/i,
+        )
+    ) {
+      // don't upload this file. it'll silently be missing
+      // @todo consider throwing an error instead here
+      return Promise.resolve();
```

Comment:
I changed from uploading the file but replacing the URL in metadata with just not uploading the file - the effect is the same (no image/icon available) but without us storing them